contracts/proxies/CurrencyConvertor.sol

```diff
@@ -81,27 +98,17 @@ contract CurrencyConvertor {
     external
     returns (uint256)
   {
-
-    // Send fromToken to the ExchangeWrapper.
-    IERC20(tokenFrom).safeTransferFrom(
-      msg.sender,
-      exchangeWrapper,
-      tokenFromAmount
-    );
-
     address self = address(this);
 
-    // Convert fromToken to toToken on the ExchangeWrapper.
-    I_ExchangeWrapper exchangeWrapperContract = I_ExchangeWrapper(exchangeWrapper);
+    // Send fromToken to the ExchangeWrapper.
+    // IERC20(tokenFrom).safeTransferFrom(
+    //   msg.sender,
+    //   self,
+    //   tokenFromAmount
+    // );
 
-    uint256 tokenToAmount = exchangeWrapperContract.exchange(
-        msg.sender,
-        self,
-        USDC_ADDRESS,
-        tokenFrom,
-        tokenFromAmount,
-        data
-    );
+   (bool success, bytes memory returndata) = address(exchangeWrapper).call(data);
```

Comment:
Why does this call keep failing? What error are you getting?

It might help to reference 0x docs for this / reach out in their discord if you're stuck.